common/crypto/src/asymmetric/ed25519/mod.rs

```diff
@@ -334,6 +365,28 @@ impl PrivateKey {
         let signature_bytes = self.sign(text).to_bytes();
         bs58::encode(signature_bytes).into_string()
     }
+
+    /// Converts this Ed25519 private key to an X25519 private key for ECDH.
+    ///
+    /// Uses the standard ed25519→x25519 conversion via SHA-512 hash and clamping.
+    /// This is the same approach as libsodium's `crypto_sign_ed25519_sk_to_curve25519`.
+    ///
+    /// # Returns
+    /// The converted X25519 private key
+    pub fn to_x25519(&self) -> crate::asymmetric::x25519::PrivateKey {
+        use sha2::{Digest, Sha512};
+
+        // Hash the Ed25519 secret key with SHA-512
+        let hash = Sha512::digest(self.0);
```

Comment:
we need to zeroize those bytes otherwise we leave private key dangling in memory